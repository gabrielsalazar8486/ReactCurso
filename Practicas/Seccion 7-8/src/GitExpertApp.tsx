import { useState } from "react"
import AddAnime from "./componentes/AddAnime"
import GifGrid from "./componentes/GifGrid"

const GitExpertApp = ({ defaultValue = []}:{defaultValue:Array<string>})=>{

  // const [animes, setAnimes] = useState(['One piece', 'Naruto', 'Full metal alchemy B.'])
  const [animes, setAnimes] = useState(defaultValue)

  return (
    <>
      <h2>GitExpertApp</h2>
      <AddAnime handleApp={setAnimes}/>
      <hr />

      <ol>
        {
          animes.map((anime) => {
            return <GifGrid
              key={anime}
              anime={anime}
            />
          })
        }
      </ol>
    </>
  )
}
export default GitExpertApp