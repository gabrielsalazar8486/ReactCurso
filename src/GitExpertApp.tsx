import { useState } from "react"
import AddAnime from "./componentes/AddAnime"
import GifGrid from "./componentes/GifGrid"

const GitExpertApp = ()=>{

  // const [animes, setAnimes] = useState(['One piece', 'Naruto', 'Full metal alchemy B.'])
  const [animes, setAnimes] = useState(['One piece'])

  return (
    <>
      <h2>GitExpertApp</h2>
      <AddAnime handleApp={setAnimes}/>
      <hr />

      <ol>
        {
          animes.map((anime) => {
            return <GifGrid
              key={anime}
              anime={anime}
            />
          })
        }
      </ol>
    </>
  )
}
export default GitExpertApp