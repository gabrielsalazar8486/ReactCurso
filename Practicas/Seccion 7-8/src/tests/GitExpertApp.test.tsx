import { shallow } from "enzyme"
import GitExpertApp from "../GitExpertApp"

describe('Pruebas en el componente GitExpertApp', ()=>{
  let wrapper = shallow(<GitExpertApp />)

  test('Se debe de mostrar corectamente el componente', ()=>{
    expect(wrapper).toMatchSnapshot()
  })

  test('Se debe de mostrar una lista de categorias', ()=>{
    const animeList = ['One piece', 'HunterXHunter']

    let wrapper = shallow(<GitExpertApp defaultValue={animeList}/>)
    expect(wrapper.find('GifGrid').length).toBe(animeList.length)
  })
})