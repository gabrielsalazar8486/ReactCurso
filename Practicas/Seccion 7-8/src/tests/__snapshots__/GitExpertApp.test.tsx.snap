// Vitest Snapshot v1

exports[`Pruebas en el componente GitExpertApp > Se debe de mostrar corectamente el componente 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddAnime
    handleApp={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente GitExpertApp > Se debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddAnime
    handleApp={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
